kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/config/Feature.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.config;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import io.kroxylicious.proxy.config.Configuration;
+
+/**
+ * Configuration Features related to how we load configuration.
+ */
+public enum Feature {
+
+    /**
+     * Enables loading specific nodes in the configuration file which configure test-only features. These are unsupported
+     * for production usages and there are no compatibility guarantees for their configuration model.
+     */
+    TEST_ONLY_CONFIGURATION {
+        @Override
+        public Optional<String> maybeWarning(boolean enabled) {
+            if (enabled) {
+                return Optional.of(
+                        "test-only configuration for proxy will be loaded. these configurations are unsupported and have no compatibility guarantees, they could be removed or changed at any time");
+            }
+            else {
+                return Optional.empty();
+            }
+        }
+
+        @Override
+        public Stream<String> supports(Configuration configuration, boolean enabled) {
```

Comment:
not sure I understand the cardinality here, why does this return a Stream of values rather than just a Optional?

Reply:
I was thinking you could find multiple reasons a configuration isn't supported.